src/lib/cli.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Main CLI object, it reads the options from cli.
+ */
+
+/*
+ * The CLI object should *not* call process.exit() directly. It should only return
+ * exit codes. This allows other programs to use the CLI object and still control
+ * when the program exits.
+ */
+
+import * as path from 'path';
+
+import { debug as d } from './utils/debug';
+import * as globby from 'globby';
+import chalk from 'chalk';
+
+const debug: debug.IDebugger = d(__filename);
+
+import { options } from './cli/options';
+import { CLIOptions, IMDFile, ILink } from './types';
+import { MDFile } from './mdfile';
+
+const getMDFiles = async (directory): Promise<IMDFile[]> => {
+    const filesPath = await globby(['**/*.md', '!node_modules', '!**/node_modules'], { cwd: directory });
+
+    return filesPath.map((relativePath) => {
+        const file = new MDFile(directory, relativePath);
+
+        return file;
+    });
+};
+
+const validateLinks = async (mdFiles: IMDFile[]): Promise<void> => {
+    for (const mdFile of mdFiles) {
+        // Validate file by file to prevent 429 and socket hang up errors.
+        await mdFile.validateLinks();
+    }
+};
+
+const getInvalidLinks = (mdFiles: IMDFile[]): ILink[] => {
+    return mdFiles.reduce((total, mdFile) => {
+        const invalidLinks = [...mdFile.invalidLinks];
+
+        return total.concat(invalidLinks);
```

Comment:
May as well combine the previous two lines with this one (which still accounts for `mdFile.invalidLinks` being a `Set`):
```suggestion
        return [...total, ...mdFile.invalidLinks];
```